source/core/src/main/com/csse3200/game/areas/ForestGameArea.java

```diff
@@ -230,165 +151,148 @@ private Entity spawnPlayer() {
-        }
+  private void spawnKangarooBoss() {
+    if (!kangarooBossSpawned) {
+      Entity kangarooBoss = NPCFactory.createKangaBossEntity(player);
+      spawnEntityOnMap(kangarooBoss);
+      kangarooBossSpawned = true;
     }
+  }
 
-  /**
-   * Spawns a chicken enemy, with the player entity as its target
-   */
-   private void spawnChicken() {
+  private void spawnEntityOnMap(Entity entity) {
     GridPoint2 minPos = new GridPoint2(PLAYER_SPAWN.x - 10, PLAYER_SPAWN.y - 10);
     GridPoint2 maxPos = new GridPoint2(PLAYER_SPAWN.x + 10, PLAYER_SPAWN.y + 10);
-
     GridPoint2 randomPos = RandomUtils.random(minPos, maxPos);
-    Entity chicken = EnemyFactory.createChicken(player);
-    enemies.add(chicken);
+    spawnEntityAt(entity, randomPos, true, true);
+  }
+
+  private void spawnItems() {
+    Supplier<Entity> generator;
 
-    float proximityRange = 0.05f; // Set a suitable proximity range
-    chicken.addComponent(new ProximityComponent(player, proximityRange)); // Add ProximityComponent
+    // Health Potions
+    generator = () -> ItemFactory.createHealthPotion(player);
+    spawnRandomItem(generator, config.spawns.NUM_HEALTH_POTIONS);
 
-    spawnEntityAt(chicken, randomPos, true, true);
+    // Apples
+    generator = () -> ItemFactory.createApple(player);
+    spawnRandomItem(generator, config.spawns.NUM_APPLES);
   }
-  /**
-   * spawns a frog enemy, with the player entity as its target
-   */
-  private void spawnFrog() {
-    GridPoint2 minPos = new GridPoint2(PLAYER_SPAWN.x - 20, PLAYER_SPAWN.y - 10);
-    GridPoint2 maxPos = new GridPoint2(PLAYER_SPAWN.x + 20, PLAYER_SPAWN.y + 10);
 
-    GridPoint2 randomPos = RandomUtils.random(minPos, maxPos);
-    Entity frog = EnemyFactory.createFrog(player);
-    enemies.add(frog);
+  private void spawnEnemies() {
+    Supplier<Entity> generator;
 
-    float proximityRange = 0.05f; // Set a suitable proximity range
-    frog.addComponent(new ProximityComponent(player, proximityRange)); // Add ProximityComponent
+    // Chicken
+    generator = () -> EnemyFactory.createChicken(player);
+    spawnRandomEnemy(generator, config.spawns.NUM_CHICKENS, 0.05);
 
-    spawnEntityAt(frog, randomPos, true, true);
+    // Monkey
+    generator = () -> EnemyFactory.createMonkey(player);
+    spawnRandomEnemy(generator, config.spawns.NUM_MONKEYS, 0.04);
 
+    // Frog
+    generator = () -> EnemyFactory.createFrog(player);
+    spawnRandomEnemy(generator, config.spawns.NUM_FROGS, 0.06);
   }
 
-  /**
-   * spawns a monkey enemy, with the player entity as its target
-   */
-  private void spawnMonkey() {
-    GridPoint2 minPos = new GridPoint2(PLAYER_SPAWN.x - 20, PLAYER_SPAWN.y - 10);
-    GridPoint2 maxPos = new GridPoint2(PLAYER_SPAWN.x + 20, PLAYER_SPAWN.y + 10);
+  private void spawnFriendlyNPCs() {
+    Supplier<Entity> generator;
 
-    GridPoint2 randomPos = RandomUtils.random(minPos, maxPos);
-    Entity monkey = EnemyFactory.createMonkey(player);
-    enemies.add(monkey);
+    // Cow
```

Comment:
Here you have removed the ability to just spawn a cow it can now only be in this certain aspect. If we need this in the future for different games etc. how would we now access this? We can't. 

I also feel that if you are changing the way FNPCs are spawning team 4 really should be involved since you are changing our stuff without our knowledge throughout the sprint in which we are still working on. Foe example my team are currently creating more entities that another team has requested and no one had communicated this change as well.

Reply:
actually brewing on it more its probably fine to do this way